tests/OptionalTest.cpp

```diff
@@ -53,6 +53,8 @@
 #endif
 // [CLI11:verbatim]
 
+TEST_CASE("OptionalNoEmpty") { CHECK(1 == 1); }
```

Comment:
Not sure if there's a nicer way to make Catch2 (maybe only 3+?) not error on no tests run in a file.

Reply:
I found `--allow-running-no-tests` but that's probably only in 3+, and we still support 2, and don't want to have to figure out how to make it optimally get added in meson (or bother with optionally adding it in CMake). So I think this is a fine workaround for now. (Plus it only affects this file, while the flag is tricky to add just for one file)

Reply:
we could either figure out how to get CMake to do the same detection and not generate that executable,  or we could just add that whole file to the optionTypeTests and delete the optionalTest file,  then it wouldn't be an issue.